e2e/app/component/workspace-card.ts

```diff
@@ -35,20 +36,31 @@ export default class WorkspaceCard extends CardBase {
   }
 
   /**
-   * Find all visible Workspace Cards. Assume at least one Card exists.
+   * Find all visible Workspace Cards.
    * @param {Page} page
    * @throws TimeoutError if fails to find Card.
    */
-  static async findAllCards(page: Page): Promise<WorkspaceCard[]> {
+  static async findAllCards(page: Page, accessLevel?: WorkspaceAccessLevel): Promise<WorkspaceCard[]> {
     try {
       await page.waitForXPath(WorkspaceCardSelector.cardRootXpath, {visible: true, timeout: 1000});
     } catch (e) {
       return [];
     }
-    const cards = await page.$x(WorkspaceCardSelector.cardRootXpath);
+    const cardElements = await page.$x(WorkspaceCardSelector.cardRootXpath);
     // transform to WorkspaceCard object
-    const resourceCards = cards.map(card => new WorkspaceCard(page).asCard(card));
-    return resourceCards;
+    const workspaceCards = cardElements.map(card => new WorkspaceCard(page).asCard(card));
+
+    const filtered = [];
+    if (accessLevel !== undefined) {
+      for (const card of workspaceCards) {
+        const cardAccessLevel = await card.getWorkspaceAccessLevel();
+        if (cardAccessLevel === accessLevel) {
+          filtered.push(card);
+        }
+      }
+      return filtered;
```

Comment:
actually, you can combine this too.

`return (await page.$x(WorkspaceCardSelector.cardRootXpath)
                .map(card => newWorkspaceCard(page).asCard(card)
                .filter(card => !accessLevel || (await card.getWorkspaceAccessLevel()) === accessLevel)`

Reply:
Good suggestion. I combined some code. But unable to use  `.filter` out of the box with async function . Googled for how to, I think I'll needs new custom code to make filter work with async function.